docs/docs/validators/partial-set-security-for-validators.md

```diff
@@ -0,0 +1,118 @@
+```bash
+interchain-security-pd query provider list-proposed-consumer-chains
+```
+
+
+:::tip
+By setting the `consumer-pub-key`, a validator can both opt in to a chain, as well as, assign a
+public key on a consumer chain. Note that a validator can always issue a [key assignment](../features/key-assignment.md)
+at a later stage to assign a new consumer public key on a chain. The key-assignment [rules](../features/key-assignment.md#rules)
+still apply when setting `consumer-pub-key` when opting in.
+:::
+
+:::info
+A validator is only eligible for consumer rewards from a consumer chain if the validator is opted in on that chain.
+:::
+
+### How to opt out from a consumer chain?
+A validator can opt out from a consumer by issuing the following message:
+
+```bash
+interchain-security-pd tx provider opt-out <consumer-chain-id>
+```
+where
+- `consumer-chain-id` is the string identifier of the consumer chain.
+
+:::warning
+A validator cannot opt out from a Top N chain if it belongs to the top N% validators on that chain.
+:::
+
+:::warning
+A validator can stop its node on a consumer chain **only** after opting out and confirming through the `has-to-validate`
+query (see [below](./partial-set-security-for-validators.md#which-chains-does-a-validator-have-to-validate)) that it does
+not have to validate the consumer chain any longer.
+:::
+
+:::warning
+If all validators opt out from an Opt-In chain, then the chain would halt.
+:::
+
+### How to set specific per consumer chain commission rate?
+A validator can choose to set a different commission rate on each of the consumer chains it validates.
+This can be done with the following command:
+```bash
+interchain-security-pd tx provider set-consumer-commission-rate <consumer-chain-id> <commission-rate>
+```
+where
+
+- `consumer-chain-id` is the string identifier of the consumer chain;
+- `comission-rate` decimal in `[minRate, 1]` where `minRate` corresponds to the minimum commission rate set on the
+provider chain (see `min_commission_rate` in `interchain-security-pd query staking params`).
+
+
+## Queries
+Partial Set Security introduces a number of queries to assist validators determine which consumer chains they have to
+validate, their commission rate per chain, etc.
+
+### Which chains does a validator have to validate?
+Naturally, a validator is aware of the Opt-In chains it has to validate because in order to validate an Opt-In chain,
+a validator has to manually opt in to the chain. This is not the case for Top N chains where a validator might be required
+to validate such a chain without explicitly opting in if it belongs to the top N% of the validators on the provider.
+
+We introduce the following query:
+```bash
+interchain-security-pd query provider has-to-validate <provider-validator-address>
+```
+that can be used by validator with `provider-validator-address` address to retrieve the list of chains that it has to validate.
```

Comment:
I would make it clear that the query is a bit forward-looking, and just emphasize something, maybe in a tip:
As a validator, this is the list of chains that you *should* be running nodes on to not get jailed for downtime.

Reply:
I would maybe replace the next sentence with this (I think the next sentence is a bit too technical to matter purely from an ops perspective)